core/redis_client.py

```
@@ -0,0 +1,443 @@
+        return await self.retry(lambda: self.pool.ltrim(key, start, end))
+
+    async def llen(self, key: str) -> int:
+        """Get length of list"""
+        return await self.retry(lambda: self.pool.llen(key))
+
+    async def scan(self, match: str, count: int = 100) -> List[str]:
+        """Scan for keys matching pattern"""
+        cursor = 0
+        results = []
+        while True:
+            cursor, keys = await self.retry(lambda: self.pool.scan(cursor=cursor, match=match, count=count))
```

Ruff (0.14.8)

Function definition does not bind loop variable `cursor`

(B023)